src/triage/predictlist/__init__.py

```diff
@@ -0,0 +1,610 @@
+        self.feature_generator.process_table_tasks(
+            self.feature_generator.generate_all_table_tasks(
+                collate_aggregations,
+                task_type='aggregation'
+            )
+        )
+        # 3. Reconstruct feature disctionary from feature_names and generate imputation
+        reconstructed_feature_dict, imputation_table_tasks = self.get_feature_dict_and_imputation_task(
+                collate_aggregations, 
+                self.retrained_model_id
+        )
+        self.feature_generator.process_table_tasks(imputation_table_tasks)
+ 
+        # 4. Build matrix
+        db_config = {
+            "features_schema_name": "triage_production",
+            "labels_schema_name": "public",
+            "cohort_table_name": cohort_table_name,
+        }
+
+        matrix_builder = MatrixBuilder(
+            db_config=db_config,
+            matrix_storage_engine=self.matrix_storage_engine,
+            engine=self.db_engine,
+            experiment_hash=None,
+            replace=True,
+        )
+        # Use timechop to get the time definition for production
+        temporal_config = self.experiment_config["temporal_config"]
+        timechopper = Timechop(**temporal_config)
+        prod_definitions = timechopper.define_test_matrices(
+            train_test_split_time=dt_from_str(prediction_date), 
+            test_duration=temporal_config['test_durations'][0],
+            test_label_timespan=temporal_config['test_label_timespans'][0]
+        )
+        
+        last_split_definition = prod_definitions[-1]
+            
+        matrix_metadata = Planner.make_metadata(
+            matrix_definition=last_split_definition,
+            feature_dictionary=reconstructed_feature_dict,
+            label_name=self.label_name,
+            label_type='binary',
+            cohort_name=self.cohort_name,
+            matrix_type='production',
+            feature_start_time=self.feature_start_time,
+            user_metadata=self.user_metadata,
+        )
+    
+        matrix_metadata['matrix_id'] = str(prediction_date) +  f'_model_id_{self.retrained_model_id}' + '_risklist'
+
+        matrix_uuid = filename_friendly_hash(matrix_metadata)
+    
+        matrix_builder.build_matrix(
+            as_of_times=[prediction_date],
+            label_name=self.label_name,
+            label_type='binary',
+            feature_dictionary=reconstructed_feature_dict,
+            matrix_metadata=matrix_metadata,
+            matrix_uuid=matrix_uuid,
+            matrix_type="production",
+        )
+        
+        # 5. Predict the risk score for production
+        predictor = Predictor(
+            model_storage_engine=self.project_storage.model_storage_engine(),
+            db_engine=self.db_engine,
+            rank_order='best'
+        )
+
+        predictor.predict(
+            model_id=self.retrained_model_id,
```

Comment:
Might want to check that this has been set by a call to `retrain()` up front and raise an error if not before doing all the other work of creating matrices, etc.?